Sprint 9

byBlinque opened the pull request.

byBlinque committed:
Search activity xml elements and logic added

byBlinque committed:
Search activity xml redesigned

byBlinque committed:
onSavedInstanceState() & onRestoreInstanceState() commited

byBlinque committed:
onSavedInstanceState() & onRestoreInstanceState() commited 2

ilshat-abdulin commented on app/src/main/java/com/example/playlistmakernew/SearchActivity.kt:
```diff
@@ -2,10 +2,72 @@ package com.example.playlistmakernew
+
+        searchET.setText(searchText)
+
+        clearButton.setOnClickListener {
+            searchET.setText("")
+        }
```
⚠ Добавь еще сюда скрытие клавиатуры. И сделай clearButton в верстке изначально gone. Сейчас если открывается экран поиска, то эта иконка сразу отображается до ввода текста в поисковике

byBlinque committed:
default styles added

byBlinque committed:
ClearTextButton visibility set on GONE
Added style parent in night theme
Commited hide keyboard feature

byBlinque committed:
1

byBlinque committed:
Не знаю почему, но в главном активити в стиле кнопок не работает iconTint - а когда устанавливаю непосредственно в самой кнопке - работает. Установил в кнопке, тк в обоих темах цвет одинаковый.

ilshat-abdulin commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -33,8 +33,9 @@
-            style="@style/MainAcButton"
             app:icon="@drawable/light_mode"
+            style="@style/MainAcButton"
+            app:iconTint="@color/yp_black"
```
🍏 В LinearLayout на 26 и 27 строках повторяющиеся размеры - в dimens есть такие размеры, можешь их здесь использовать. И можно вместо android:layout_marginLeft и  android:layout_marginRight использовать layout_marginHorizontal

ilshat-abdulin commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,83 @@
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:orientation="horizontal"
+        android:layout_marginStart="20dp"
+        android:layout_marginTop="20dp">
```
⚠ Также размеры повторяются. Лучше стараться привыкать подобные вещи определять в ресурсах

ilshat-abdulin commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,83 @@
+            android:orientation="horizontal"
+            style="@style/SearchBarBGStyle">
+            <ImageView
+                android:layout_width="15dp"
+                android:layout_height="15dp"
```
⚠Также повторяются. Стоит пройтись по всей верстке и привести в порядок. 
Размеры рекомендуется использовать кратны 4 и 8, несмотря на то, что в дизайне могут быть нечетные и несимметричные размеры

ilshat-abdulin commented on app/src/main/res/values-v26/styles.xml:
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="SettingsAcMenuItemWrapper">
+        <item name="android:layout_height">@dimen/settings_menu_item_height_size</item>
+        <item name="android:paddingHorizontal">@dimen/settings_menu_item_padding_horizontal</item>
+    </style>
+
+    <style name="MainAcButton">
+        <item name="android:backgroundTint">@color/white</item>
+        <item name="android:textColor">@color/text_color</item>
+        <item name="android:textSize">@dimen/main_button_text_size</item>
+        <item name="cornerRadius">@dimen/main_button_corner_radius</item>
+        <item name="iconGravity">textStart</item>
+        <item name="iconPadding">@dimen/main_button_icon_padding</item>
+        <item name="android:iconTint">@color/text_color</item>
+        <item name="android:fontFamily">@font/ys_display_medium</item>
+        <item name="android:textAllCaps">false</item>
+        <item name="android:textStyle">bold</item>
+        <item name="android:letterSpacing">0</item>
+    </style>
```
⚠ Вот этот файл лучше удалить. Это касаемо того, что iconTint не работает.

ilshat-abdulin commented:
Евгений, вопросы можешь задавать здесь, а не в коммите). iconTint заработает в стилях, если удалишь лишний файл styles в  пакете values-v26. Ну или он еще отображается как styles.xml (v26). В нем остался стиль, который ломает настройку с iconTint. 

Еще рекомендую исправить еще один момент, который я не доглядел изначально, но вынужден все же попросить тебя:
Обрати внимание, как выглядит иконка поддержки. Экспортирована только часть иконки
![Снимок экрана 2023-02-18 133251](https://user-images.githubusercontent.com/72722086/219850578-d7f5a5e9-d3ac-4736-9a08-667f6b318004.png)

Когда вытаскиваешь иконки из дизайна, то нужно обращать внимание, какой слой иерархии захватываешь. Должно быть так

![Снимок экрана 2023-02-18 131647](https://user-images.githubusercontent.com/72722086/219850621-e58a0f08-e688-40b1-b9b3-f6e3a6a37882.png)

ilshat-abdulin commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -1,9 +1,83 @@
+            android:textSize="@dimen/text_header_size"
+            android:layout_marginLeft="28dp"
+            style="@style/SearchToolbarColor" />
+    </LinearLayout>
+    <com.google.android.material.card.MaterialCardView
```
🍏 Между компонентами оставляй отступы или пользуйся автоформатированием - студия добавит отступы, подкорректирует выравнивание 
Комбинация клавиш:
Windows: Ctrl + Alt + L
Mac: Option + Command + L

ilshat-abdulin commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,62 @@
+        <item name="iconTint">@color/text_color</item>
+        <item name="android:fontFamily">@font/ys_display_medium</item>
+        <item name="android:textAllCaps">false</item>
+        <item name="android:textStyle">bold</item>
```
⚠ По дизайну стиль шрифта задан, как bold

byBlinque committed:
Убрал повторяющиеся размеры
Удалил файл values-v26
Нормально импортировал иконку с оператором
Добавил отступы в xml layout-ах
Поменял шрифт согласно верстке

byBlinque committed:
Подправил шрифты согласно макету

ilshat-abdulin commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -1,101 +1,112 @@
-            style="@style/SettingsAcMenuItemIcon"
             android:maxWidth="16dp"
-            />
+            android:src="@drawable/share" />
```
🍏 Теперь, после добавления обновленной новой иконки @drawable/support_icon_final, появились небольшие различия в отступах. Это видно, если сравнить support_icon_final с иконками @drawable/share и @drawable/arrow_forward. 
Это происходит  потому, что support_icon_final теперь имеет размер 24х24, в отличии от share(16х18) и arrow_forward(8х14). 
Если наблюдается подобное, то стоит и остальные иконки экспортировать с размером 24х24, выбрав правильную иерархию слоя в дизайне.

Это не будет считаться ошибкой, но лучше исправить - делов на 5 минут

byBlinque replied:
Блин, спасибо тебе за такие развернутые ответы! Еще с прошлого раза приметил это. Написал о тебе отзыв в конце спринта) На счет иконок я ломал голову как сделать нормальные отступы (не увидел слои в фигме, импортировал то, что смог накликать).

ilshat-abdulin replied:
Рад помочь. Спасибо большое за отзыв)

byBlinque committed:
Добавил иконки нормального размера (24х24)

byBlinque committed:
Заменил на иконки нормального размера в activity_settings.xml
